src/NUnitEngine/nunit.engine.netstandard/Services/DefaultTestRunnerFactory.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using NUnit.Engine.Internal;
+using NUnit.Engine.Runners;
+
+namespace NUnit.Engine.Services
+{
+    /// <summary>
+    /// DefaultTestRunnerFactory handles creation of a suitable test 
+    /// runner for a given package to be loaded and run either in a 
+    /// separate process or within the same process. 
+    /// </summary>
+    public class DefaultTestRunnerFactory : ITestRunnerFactory
+    {
+        #region InProcessTestRunnerFactory Overrides
+
+        /// <summary>
+        /// Returns a test runner based on the settings in a TestPackage.
+        /// Any setting that is "consumed" by the factory is removed, so
+        /// that downstream runners using the factory will not repeatedly
+        /// create the same type of runner.
+        /// </summary>
+        /// <param name="package">The TestPackage to be loaded and run</param>
+        /// <returns>A TestRunner</returns>
+        public virtual ITestEngineRunner MakeTestRunner(TestPackage package)
+        {
+            int assemblyCount = 0;
+
+            foreach (var subPackage in package.SubPackages)
+            {
+                var testFile = subPackage.FullName;
+
+                if (PathUtils.IsAssemblyFileType(testFile))
+                    assemblyCount++;
+            }
+
+            if ( assemblyCount > 0)
+                return new AggregatingTestRunner(package);
+
+            // TODO: What about bad extensions?
+            return new DirectTestRunner(package);
+        }
```

Comment:
Agreed, but I didn't want to strip out too much that we might end up re-adding. We don't need NUnit project loading or solution loading for the VS Adapter, but I thought that once we move to other runners like Xamarin or UWP, we might end up adding some of the extensions back in.